src/Compilers/Core/Portable/DiagnosticAnalyzer/CompilerDiagnosticAnalyzer.CompilationAnalyzer.cs

```diff
@@ -116,7 +120,11 @@ public override int GetHashCode()
 
                 public override bool Equals(Diagnostic? obj)
                 {
-                    return _original.Equals(obj);
+                    if (obj instanceof CompilerDiagnostic cd) {
+                        return _original.Equals(cd._original);
```

Comment:
@RikkiGibson we'll still need this.

Reply:
Don't we also need to check that the "properties" on each instance are equal?

Reply:
The _original.Equals() check already does that.